dotenv-linter/src/checks/quote_character.rs

```diff
@@ -31,6 +31,11 @@ impl Check for QuoteCharacterChecker<'_> {
             return None;
         }
 
+        // Don't remove quotes for arrays of JSON strings, e.g.: '["a"]'
+        if val.contains('\'') && val.contains('\"') && val.contains('[') && val.contains(']') {
```

Comment:
Sorry, by I'm not sure of this logic to be honest  🤔 

It doesn't guarantee that line is correct JSON, it might be something like `'...]"['` and this check will ignore this?
Is there a similar behavior in other env linters or libs? It seems too specific to cover JSON arrays separately

If we would like to support such cases - it should be somehow parsed to [serde_json::Value](https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html) for example 🤔 

However, it might be betterto disable  `QuoteCharacterChecker` for specific lines  in such cases instead of supporting all possible cases.
I'm even thinking if it should be optional checker, because it's not really breaks ENV format in any way and users might want to wrap values into `'`/`'"` due to different reasons.

@dotenv-linter/core WDYT?

Reply:
My naive view is that we should disable the `QuoteCharacterChecker` in such cases as we otherwise would run into a situation where we would need to support more and more special cases for other formats.

Reply:
> However, it might be better to disable QuoteCharacterChecker for specific lines in such cases instead of supporting all possible cases on linter side.

I agree. Do I understand correctly that the user himself should disable this check using the control comments?
```
# dotenv-linter:off QuoteCharacter
FOO='["a","b"]'
```

Reply:
Yes, I meant this way